pkgs/tools/package-management/nix/common.nix

```diff
@@ -291,6 +283,16 @@ self = stdenv.mkDerivation {
         inherit lib pkgs;
         nix = self;
       };
+    } // lib.optionalAttrs stdenv.isLinux {
+      nixStatic = pkgsStatic.nixVersions.${self_attribute_name};
+
+      # Basic smoke tests that needs to pass when upgrading nix.
+      # Note that this test does only test the nixVersions.stable attribute.
+      misc = nixosTests.nix-misc.default;
+      upgrade = nixosTests.nix-upgrade;
+      simpleUefiGrub = nixosTests.installer.simpleUefiGrub;
```

Comment:
Can we use `simpleUefiSystemdBoot` here like the channels do, rather than proliferating use of an unmaintained bootloader? The bootloader shouldn’t matter for testing Nix.